journals/apps/journals/models.py

```diff
@@ -683,37 +688,49 @@ def get_context(self, request, *args, **kwargs):
 
         return context
 
-    def get_prev_page(self):
+    def get_prev_page(self, live_only=True):
         """
         Get the previous page for navigation. Search order is previous sibling's last descendant,
         previous sibling, then parent
         """
-        prev_sib = self.get_prev_sibling()
+        prev_sib = self.get_prev_siblings().first()
         if prev_sib:
-            last_child = prev_sib.specific.get_last_descendant()
+            last_child = prev_sib.specific.get_last_descendant(live_only=live_only)
             return last_child if last_child else prev_sib
 
-        parent = self.get_parent()
-        return parent if parent and isinstance(parent.specific, JournalPage) else None
+        prev_ancestor = self.get_ancestors().last()
+        if prev_ancestor and isinstance(prev_ancestor.specific, JournalPage):
+            return prev_ancestor if \
```

Comment:
This is a little hard to read and has some duplicate logic. 
- Ternary expressions save us extra lines, but if you need the lines anyway, best to just use a regular if statement. 
- `(not live_only or live_only and prev_ancestor.specific.live)` will always return True since the `or` will be calculated before the `and`. Assuming you mean `(prev_ancestor.specific.live and live_only or not live_only)` this can be simplified down to `prev_ancestor.specific.live or not live_only` because `A * B + ~B` == `A + ~B`. Suggested replacement:
```
if prev_ancestor and isinstance(prev_ancestor.specific, JournalPage):
    if prev_ancestor.specific.live or not live_only:
        return prev_ancestor
    return prev_ancestor.specific.get_prev_page(live_only=live_only)
return None
```

Reply:
fixed